clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/ClusteredStoreTest.java

```diff
@@ -143,4 +143,42 @@ public void testClear() throws Exception {
     assertThat(store.containsKey(2L), is(false));
     assertThat(store.containsKey(3L), is(false));
   }
+
+  @Test
+  public void testPutIfAbsent() throws Exception {
+    assertThat(store.putIfAbsent(1L, "one"), nullValue());
+    validateStat(store, StoreOperationOutcomes.PutIfAbsentOutcome.PUT, 1);
+    assertThat(store.putIfAbsent(1L, "another one").value(), is("one"));
+    validateStat(store, StoreOperationOutcomes.PutIfAbsentOutcome.HIT, 1);
+  }
+
+  @Test
+  public void testConditionalRemove() throws Exception {
+    assertThat(store.remove(1L, "one"), is(Store.RemoveStatus.KEY_MISSING));
+    store.put(1L, "one");
+    assertThat(store.remove(1L, "one"), is(Store.RemoveStatus.REMOVED));
+    store.put(1L, "another one");
+    assertThat(store.remove(1L, "one"), is(Store.RemoveStatus.KEY_PRESENT));
+    validateStat(store, StoreOperationOutcomes.ConditionalRemoveOutcome.REMOVED, 1);
+    validateStat(store, StoreOperationOutcomes.ConditionalRemoveOutcome.MISS, 2);
+  }
+
+  @Test
+  public void testReplace() throws Exception {
+    assertThat(store.replace(1L, "one"), nullValue());
+    store.put(1L, "one");
+    assertThat(store.replace(1L, "another one").value(), is("one"));
+    validateStat(store, StoreOperationOutcomes.ReplaceOutcome.REPLACED, 1);
```

Comment:
This _bulk_ validation of stats is why you missed the inverted result in code. Make sure you validate stats immediately after the operation invocation instead.

Applies to all added tests.